plan/statistics/statistics.go

```diff
@@ -186,11 +187,75 @@ func (c *Column) search(sc *variable.StatementContext, target types.Datum) (inde
 	return
 }
 
+// mergeBuckets is used to merge every two neighbor buckets.
+func (c *Column) mergeBuckets(bucketIdx int64) {
+	curBuck := 0
+	for i := int64(0); i+1 <= bucketIdx; i += 2 {
+		c.Numbers[curBuck] = c.Numbers[i+1]
```

Comment:
A bucket number is the number of items stored in all previous buckets and the current bucket.